server/server.go

```diff
@@ -943,3 +946,57 @@ func isNotFound(err error) bool {
 
 	return false
 }
+
+// startReloadWatcherForMirrorRegistries starts a new goroutine that
+// monitors the "registries.conf.d" directory. If not set, it defaults to
+// "/etc/containers/registries.conf.d". The routine reloads the registries
+// configuration when a file is modified or created in the directory.
+func (s *Server) startReloadWatcherForMirrorRegistries(ctx context.Context, registriesConfDDir string) {
+	if registriesConfDDir == "" {
+		log.Infof(ctx, "No registries.conf.d directory specified, using default")
+		registriesConfDDir = "/etc/containers/registries.conf.d"
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf(ctx, "Failed to create new watch: %v", err)
+		return
+	}
+	defer watcher.Close()
+
+	log.Infof(ctx, "Registered reload watcher for mirror registries configuration")
+	done := make(chan struct{})
+	go s.reloadMirrorRegistriesConfiguration(ctx, watcher, done, registriesConfDDir)
+	<-done
+}
+
+func (s *Server) reloadMirrorRegistriesConfiguration(ctx context.Context, watcher *fsnotify.Watcher, done chan struct{}, path string) {
+	defer close(done)
+	if err := watcher.Add(path); err != nil {
+		log.Errorf(ctx, "Watcher.Add(%q) failed: %s", path, err)
+		return
+	}
+
+	for {
+		select {
+		case event, ok := <-watcher.Events:
+			if !ok {
+				return
+			}
+			// Skip events for files starting with a dot
+			if strings.HasPrefix(filepath.Base(event.Name), ".") {
+				continue
+			}
+			if event.Op&(fsnotify.Write|fsnotify.Create) != 0 {
+				log.Infof(ctx, "File %s modified or created, reloading registries configuration", event.Name)
+				if err := s.config.ReloadRegistries(); err != nil {
+					log.Errorf(ctx, "Failed to reload registry configuration: %v", err)
```

Comment:
You do like to use `%v` for errors, where we should rely on `%s` to ensure that the type `String()` method is called.

```suggestion
					log.Errorf(ctx, "Failed to reload registry configuration: %s", err)
```